sklearn/linear_model/glm.py

```diff
@@ -0,0 +1,1813 @@
+        always excluded from counting).
+
+    P2 : {None, array-like, sparse matrix}, shape \
+            (n_features*, n_features*), optional (default=None)
+        With this square matrix the L2 penalty is calculated as `w P2 w`.
+        This gives a fine control over this penalty (Tikhonov
+        regularization).
+        The default value ``None`` is the same as the idendity matrix.
+        Note that n_features* = X.shape[1] = length of coef_ (intercept
+        always excluded from counting). P2 must be positive semi-definite.
+
+    fit_intercept : boolean, optional (default=True)
+        Specifies if a constant (a.k.a. bias or intercept) should be
+        added to the linear predictor (X*coef+intercept).
+
+    family : {'normal', 'poisson', 'gamma', 'inverse.gaussian'} or an instance\
+            of class ExponentialDispersionModel, optional(default='normal')
+        the distributional assumption of the GLM, i.e. which loss function to
+        be minimized.
+
+    link : {'identity', 'log'} or an instance of class Link,
+        optional (default='identity')
+        the link function of the GLM, i.e. mapping from linear predictor
+        (X*coef) to expectation (mu).
+
+    fit_dispersion : {None, 'chisqr', 'deviance'}, optional (defaul=None)
+        method for estimation of the dispersion parameter phi. Whether to use
+        the chi squared statisic or the deviance statistic. If None, the
+        dispersion is not estimated.
+
+    solver : {'auto', 'irls', 'newton-cg', 'lbfgs', 'cd'}, \
+            optional (default='auto')
+        Algorithm to use in the optimization problem.
+
+        - 'auto' sets 'irls' if l1_ratio equals 0, else 'cd'.
+
+        - 'irls' is iterated reweighted least squares (Fisher scoring).
+            It is the standard algorithm for GLMs. Cannot deal with
+            L1 penalties.
+
+        - 'newton-cg', 'lbfgs'. Cannot deal with L1 penalties.
+
+        - 'cd' is the coordinate descent algorithm. It can
+            deal with L1 as well as L2 penalties.
+
+    max_iter : int, optional (default=100)
+        The maximal number of iterations for solver algorithms.
+
+    tol : float, optional (default=1e-4)
+        Stopping criterion. For the irls, newton-cg and lbfgs solvers,
+        the iteration will stop when ``max{|g_i | i = 1, ..., n} <= tol``
+        where ``g_i`` is the i-th component of the gradient (derivative of
+        the deviance).
+
+    warm_start : boolean, optional (default=False)
+        If set to ``True``, reuse the solution of the previous call to fit as
+        initialization for ``coef_`` and ``intercept_`` (supersedes option
+        ``start_params``). If set to ``True`` or if the attribute ``coef_``
+        does not exit (first call to fit), option ``start_params`` sets the
+        starting values for ``coef_`` and ``intercept_``.
+
+    start_params : {None, 'least_squares', 'zero', array of shape \
```

Comment:
We usually provide warm_start, but not this level of direct control over initialisation...

Reply:
Having good start values for coef_ has a huge impact on the performance of the fit.